### ISSUE TITLE
Enabling `integrity` causes script tag `async` attributes to output as `async="undefined"` (in production mode only)

### ISSUE THREAD

**Issue by davidmurdoch** [webdiscus/html-bundler-webpack-plugin#105]
## Current behaviour

In `mode: "production"` only, enabling `integrity` causes script tag `async` attributes to output as `async="undefined"`.

this line:

```
<script async src="script.js"></script>
```

is output as:

```
<script async="undefined" src="script.js" integrity="sha384-ym/3cF/gAcW1/eerNQGFaOSOuU2LHI6RYvChaVur/G+PkVozZ5Zo8Uu5R2S2zm7C" crossorigin="anonymous"></script>
```

## Expected behaviour

```
<script async src="script.js"></script>
```

should output as:

```
<script async src="script.js" integrity="sha384-ym/3cF/gAcW1/eerNQGFaOSOuU2LHI6RYvChaVur/G+PkVozZ5Zo8Uu5R2S2zm7C" crossorigin="anonymous"></script>
```

### Reproduction

create a new npm project and install dependencies:

```bash
npm init -y && npm install webpack webpack-cli html-bundler-webpack-plugin
```

Create these files:

`./index.html`: 
```html
<!DOCTYPE html>
<html>
<head>
    <title>Hello</title>
</head>
<body>
    <script async src="script.js"></script>
</body>
</html>
```

`./script.js`:
```javascript
console.log("hello");
```

`./webpack.config.js`:
```javascript
const HtmlBundlerWebpackPlugin = require('html-bundler-webpack-plugin');

module.exports = {
    mode: "production",
    entry: './index.html',
    plugins: [
        new HtmlBundlerWebpackPlugin({
            integrity: 'auto',
        })
    ],
    output: {
        crossOriginLoading: 'anonymous'
    }
}
```

run `npx -y webpack` to generate the `dist` folder.

You'll find the generated `./dist/main.html` file will contain the `async="undefined"` bug.

## Environment

- OS: [macOS, Linux, Windows] Linux
- version of Node.js: 20.11.1
- version of Webpack: 5.93.0
- version of the Plugin: 3.17.0

## Appreciation for the useful project

- [ ] After the problem is resolved, do not forget to give a star ⭐

**Comment by webdiscus** [comment]
Hello @davidmurdoch 

Thanks for reporting the issue.
I will fix it.

**Comment by webdiscus** [comment]
@davidmurdoch 

the issue is fixed in the `v3.17.1`, please test it

**Comment by davidmurdoch** [comment]
I've verified that this fixes it!
